client/src/helpers/qnAHelper.js

```diff
@@ -31,52 +31,193 @@ class QnAClientHelpers {
+        if (answer.answerer_name === 'Seller') {
+          return answer
+        }
+      })
+      if (filterSeller.length) {
 
+        sortedArr = filterSeller.concat(sortedArr)
       }
+      final.push(sortedArr)
+      count --
+      this.sortAnswers(answers, count, final)
+
+    }
+    return final
+  }
+
+  filterAll(currentQuestions) {
+    let filteredQuestions = this.sortQuestions(currentQuestions);
+
+    //get answers
+    let answers = []
+    currentQuestions.forEach((question) => {
+
+
+      answers.push(question.answers)
+
     })
 
-    let countArr = [];
-    filteredOutSeller.forEach((answer) => {
-      countArr.push(answer.helpfulness)
+    let answerVals = []
+
+    filteredQuestions.forEach((question) => {
+      let values = Object.values(question.answers)
 
+      answerVals.push(values)
     })
 
+    let filteredAnswers = this.sortAnswers(answerVals)
+
+
+    return [filteredQuestions, filteredAnswers]
+
+  }
+
+  filterSearchInput(currentQuestions, searchTerm) {
+
+    let newQuestions = currentQuestions.filter((question) => {
+      let lowerCaseQuestion = question.question_body.toLowerCase().replace(/\s+/g, '')
+      let lowerCaseInput = searchTerm.toLowerCase().replace(/\s+/g, '')
+
+      if (lowerCaseQuestion.includes(lowerCaseInput)) {
+        return question
+
+      }
 
-    countArr = countArr.sort((a, b) => (b - a));
-    countArr.forEach((number) => {
-      filteredOutSeller.forEach((answer) => {
-         if (number === answer.helpfulness) {
-           sortedArr.push(answer)
-         }
-      })
     })
 
+    return newQuestions
+  }
 
-    let filterSeller = answers.filter(answer => {
-      if (answer.answerer_name === 'Seller') {
-        return answer
+  showMoreAnsweredQuestions(arr) {
+    let showButton;
+      if (arr[0].length) {
+        showButton = true;
+      } else {
+        showButton = false
       }
-    });
+      return showButton
+  }
+
+  showMoreAnsweredBtnClass(bool, qClickCount, index) {
+    let newClass;
+    if (bool) {
+      newClass = 'moreAnsweredBtn'
+    }
+
+    if (qClickCount === index|| qClickCount - 1 === index) {
+      // console.log('hi')
+      newClass = 'moreAnswerBtn Hide'
+    }
+
+    return newClass
+
+  }
 
+  loadAnswerButtonText(currentCount) {
+    let text;
 
+    if (currentCount % 2 !== 0) {
+      text = 'Collapse Answers'
 
-   if (filterSeller.length) {
+    } else {
+      text = 'Load More Answers'
+    }
+    return text
+  }
+
+  qListScrollClass (count) {
+    let newClass;
+    if (count > 1) {
+      newClass = 'questionList scroll container'
+    }
+    return newClass
 
-     sortedArr = filterSeller.concat(sortedArr)
-   }
-   return sortedArr
+  }
 
+  answerHideClass(classname, index) {
+    let newClass
+    if (index <= 1) {
+      newClass = 'answerListTable'
+    }
+    if (classname === 'Hide') {
+      newClass = 'answerListTable Hide'
+    }
+
+    if (index >= 2) {
+      newClass = 'answerListTable Hide'
```

Comment:
You could maybe combine these two if statements into one with the or operator because they have the same result. `if (classname === 'Hide' || index >= 2) {
  newClass = 'answerListTable Hide'
}`